aten/src/ATen/native/UnaryOps.cpp

```diff
@@ -86,80 +139,80 @@ Tensor& bitwise_not_out(Tensor& result, const Tensor& self) { return unary_op_im
 Tensor bitwise_not(const Tensor& self) { return unary_op_impl(self, at::bitwise_not_out); }
 Tensor& bitwise_not_(Tensor& self) { return unary_op_impl_(self, at::bitwise_not_out); }
 
-Tensor& ceil_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, ceil_stub); }
-Tensor ceil(const Tensor& self) { return unary_op_impl(self, at::ceil_out); }
-Tensor& ceil_(Tensor& self) { return unary_op_impl_(self, at::ceil_out); }
+Tensor& ceil_out(Tensor& result, const Tensor& self) { return unary_floating_ufunc_op_impl_out(result, self, ceil_stub); }
+Tensor ceil(const Tensor& self) { return unary_floating_ufunc_op_impl(self, at::ceil_out); }
+Tensor& ceil_(Tensor& self) { return unary_floating_ufunc_op_impl_(self, at::ceil_out); }
 
-Tensor& expm1_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, expm1_stub); }
-Tensor expm1(const Tensor& self) { return unary_op_impl(self, at::expm1_out); }
-Tensor& expm1_(Tensor& self) { return unary_op_impl_(self, at::expm1_out); }
+Tensor& expm1_out(Tensor& result, const Tensor& self) { return unary_floating_ufunc_op_impl_out(result, self, expm1_stub); }
+Tensor expm1(const Tensor& self) { return unary_floating_ufunc_op_impl(self, at::expm1_out); }
+Tensor& expm1_(Tensor& self) { return unary_floating_ufunc_op_impl_(self, at::expm1_out); }
 
 Tensor& frac_out(Tensor& result, const Tensor& self) { return unary_op_impl_out(result, self, frac_stub); }
```

Comment:
frac is a composite that can alternatively be implemented as 

`(self - torch.floor(torch.abs(self)) * torch.sign(self))`

The call to floor should promote to floating. Seems like frac should be a unary floating ufunc, too.

Reply:
Added frac as unary floating ufunc. It's not there in the ONNX operator lists (https://github.com/onnx/onnx/blob/master/docs/Operators.md) - so no changes needed there. Also added in the JIT `shape_analysis.cpp` file as unary floating ufunc.